{
    "issue": {
        "title": "[C++] Segfault in json-integration-test",
        "body": "***Note**: This issue was originally created as [ARROW-3092](https://issues.apache.org/jira/browse/ARROW-3092). Please see the [migration documentation](https://gist.github.com/toddfarmer/12aa88361532d21902818a6044fda4c3) for further details.*\n\n### Original Issue Description:\nI've upgraded to Ubuntu 18.04.1 and now I get segfaults in json-integration-test:\r\n\r\n```Java\n\r\n(gdb) run\r\nStarting program: /home/antoine/arrow/cpp/build-test/debug/json-integration-test \r\n[Thread debugging using libthread_db enabled]\r\nUsing host libthread_db library \"/lib/x86_64-linux-gnu/libthread_db.so.1\".\r\n[==========] Running 2 tests from 1 test case.\r\n[----------] Global test environment set-up.\r\n[----------] 2 tests from TestJSONIntegration\r\n[ RUN      ] TestJSONIntegration.ConvertAndValidate\r\n\r\nProgram received signal SIGSEGV, Segmentation fault.\r\nstd::string::_Rep::_M_is_leaked (this=this@entry=0xffffffffffffffe8)\r\n    at /home/msarahan/miniconda2/conda-bld/compilers_linux-64_1507259624353/work/.build/x86_64-conda_cos6-linux-gnu/build/build-cc-gcc-final/x86_64-conda_cos6-linux-gnu/libstdc++-v3/include/bits/basic_string.h:3075\r\n3075\t/home/msarahan/miniconda2/conda-bld/compilers_linux-64_1507259624353/work/.build/x86_64-conda_cos6-linux-gnu/build/build-cc-gcc-final/x86_64-conda_cos6-linux-gnu/libstdc++-v3/include/bits/basic_string.h: Aucun fichier ou dossier de ce type.\r\n(gdb) bt\r\n#0  std::string::_Rep::_M_is_leaked (this=this@entry=0xffffffffffffffe8)\r\n    at /home/msarahan/miniconda2/conda-bld/compilers_linux-64_1507259624353/work/.build/x86_64-conda_cos6-linux-gnu/build/build-cc-gcc-final/x86_64-conda_cos6-linux-gnu/libstdc++-v3/include/bits/basic_string.h:3075\r\n#1  0x00007ffff7311856 in std::string::_Rep::_M_grab (this=0xffffffffffffffe8, __alloc1=..., __alloc2=...)\r\n    at /home/msarahan/miniconda2/conda-bld/compilers_linux-64_1507259624353/work/.build/x86_64-conda_cos6-linux-gnu/build/build-cc-gcc-final/x86_64-conda_cos6-linux-gnu/libstdc++-v3/include/bits/basic_string.h:3126\r\n#2  0x00007ffff731189d in std::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string (this=0x7fffffffcf68, __str=...)\r\n    at /home/msarahan/miniconda2/conda-bld/compilers_linux-64_1507259624353/work/.build/x86_64-conda_cos6-linux-gnu/build/build-cc-gcc-final/x86_64-conda_cos6-linux-gnu/libstdc++-v3/include/bits/basic_string.tcc:613\r\n#3  0x00000000005f63fd in boost::filesystem::path::path (this=0x7fffffffcf68, p=...)\r\n    at /home/antoine/miniconda3/envs/pyarrow/include/boost/filesystem/path.hpp:137\r\n#4  0x00000000005f628a in boost::filesystem::operator/ (lhs=..., rhs=...)\r\n    at /home/antoine/miniconda3/envs/pyarrow/include/boost/filesystem/path.hpp:792\r\n#5  0x00000000005f1d37 in arrow::ipc::temp_path () at ../src/arrow/ipc/json-integration-test.cc:233\r\n#6  0x00000000005f3038 in arrow::ipc::TestJSONIntegration::mkstemp (this=<error reading variable: Cannot access memory at address 0x7f00706d7427>)\r\n    at ../src/arrow/ipc/json-integration-test.cc:241\r\nBacktrace stopped: previous frame inner to this frame (corrupt stack?)\r\n```",
        "created_at": "2018-08-20T08:21:55.000Z",
        "updated_at": "2018-08-20T18:11:02.000Z",
        "labels": [
            "Migrated from Jira",
            "Component: C++",
            "Type: bug"
        ],
        "closed": true,
        "closed_at": "2018-08-20T18:11:02.000Z"
    },
    "comments": [
        {
            "created_at": "2018-08-20T15:07:49.833Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-3092?focusedCommentId=16586056) by Wes McKinney (wesm):*\nI think this is a compiler toolchain issue. If you build without an external toolchain and using the system boost libraries, does it still fail?"
        },
        {
            "created_at": "2018-08-20T16:10:10.385Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-3092?focusedCommentId=16586155) by Antoine Pitrou (apitrou):*\nIt seems that my lastest local changes (perhaps `-D_GLIBCXX_USE_CXX11_ABI=1`) had the side effect of fixing this issue, though I don't understand why."
        },
        {
            "created_at": "2018-08-20T17:49:30.891Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-3092?focusedCommentId=16586295) by Wes McKinney (wesm):*\nWhat's the last time you updated your toolchain? I think Anaconda has moved to the gcc5 ABI recently"
        },
        {
            "created_at": "2018-08-20T17:50:56.100Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-3092?focusedCommentId=16586297) by Antoine Pitrou (apitrou):*\nAh, well, I recreated my conda environment today in an attempt to fix other issues :) That might explain why."
        },
        {
            "created_at": "2018-08-20T17:53:08.909Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-3092?focusedCommentId=16586306) by Wes McKinney (wesm):*\nOur build instuctions in http://arrow.apache.org/docs/python/development.html are likely to have grown stale given the toolchain changes in recent months \r\n\r\nI opened ARROW-3096 about having a look at that. "
        },
        {
            "created_at": "2018-08-20T18:10:55.343Z",
            "body": "***Note**: [Comment](https://issues.apache.org/jira/browse/ARROW-3092?focusedCommentId=16586334) by Antoine Pitrou (apitrou):*\nCan't reproduce anymore after I have recreated my environment and changed some build variables."
        }
    ]
}